components/layout_2020/flow/inline.rs

```diff
@@ -1253,7 +1254,13 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
         let strut_size = if using_fallback_font {
             // TODO(mrobinson): This value should probably be cached somewhere.
             let container_state = self.current_inline_container_state();
-            let mut block_size = container_state.get_block_size_contribution(&font_metrics);
+            let mut block_size = container_state.get_block_size_contribution(
+                effective_vertical_align(
+                    &container_state.style,
+                    self.inline_box_state_stack.last().map(|c| &c.base),
+                ),
```

Comment:
I think this will be slightly easier to read if you save `vertical_align` to a temporary.